vkubelet/vkubelet.go

```diff
@@ -425,6 +429,73 @@ func addPodAttributes(span *trace.Span, pod *corev1.Pod) {
 	)
 }
 
+func (s *Server) onAddPod(ctx context.Context, obj interface{}) {
+	pod := obj.(*corev1.Pod)
+
+	logger := log.G(ctx).WithField("method", "onAddPod")
+	if pod == nil {
+		logger.Error("obj is not a valid pod:", obj)
+		return
+	}
+
+	logger.Debugf("Receive added pod '%s/%s' ", pod.GetNamespace(), pod.GetName())
+
+	if s.resourceManager.UpdatePod(pod) {
+		s.podCh <- pod
+	}
+}
+
+func (s *Server) onUpdatePod(ctx context.Context, obj interface{}) {
+	pod := obj.(*corev1.Pod)
+
+	logger := log.G(ctx).WithField("method", "onUpdatePod")
+	if pod == nil {
+		logger.Error("obj is not a valid pod:", obj)
+		return
+	}
+
+	logger.Debugf("Receive updated pod '%s/%s'", pod.GetNamespace(), pod.GetName())
+
+	if s.resourceManager.UpdatePod(pod) {
+		s.podCh <- pod
+	}
+}
+
+func (s *Server) onDeletePod(ctx context.Context, obj interface{}) {
+	pod := obj.(*corev1.Pod)
+
+	logger := log.G(ctx).WithField("method", "onDeletePod")
+	if pod == nil {
+		logger.Errorf("obj is not a valid pod: %v", obj)
+		return
+	}
+
+	logger.Debugf("Receive deleted pod '%s/%s'", pod.GetNamespace(), pod.GetName())
+
+	if s.resourceManager.DeletePod(pod) {
+		s.podCh <- pod
+	}
+}
+
+func (s *Server) startPodSynchronizer(ctx context.Context, id int) {
+	logger := log.G(ctx).WithField("podSynchronizer", id)
+	logger.Debug("Start pod synchronizer")
+
+	for pod := range s.podCh {
```

Comment:
Doesn't looks like anything is closing this channel.
We should either make sure it gets closed and/or use `for { select {}}` to select on `<-ctx.Done()` and `<-podCh`